dask/dataframe/io/sql.py

```diff
@@ -211,3 +213,173 @@ def _read_sql_chunk(q, uri, meta, engine_kwargs=None, **kwargs):
         return meta
     else:
         return df.astype(meta.dtypes.to_dict(), copy=False)
+
+
+def to_sql(
+    df,
+    name: str,
+    uri: str,
+    schema=None,
+    if_exists: str = "fail",
+    index: bool = True,
+    index_label=None,
+    chunksize=None,
+    dtype=None,
+    compute=True,
+    parallel=False,
+    **kwargs
+):
+    """ Store Dask Dataframe to a SQL table
+
+    An empty table is created based on the "meta" DataFrame (and conforming to the caller's "if_exists" preference), and
+    then each block calls pd.DataFrame.to_sql (with `if_exists="append"`).
+
+    Databases supported by SQLAlchemy [1]_ are supported. Tables can be
+    newly created, appended to, or overwritten.
+
+    Parameters
+    ----------
+    name : str
+        Name of SQL table.
+    uri : string
+        Full sqlalchemy URI for the database connection
+    schema : str, optional
+        Specify the schema (if database flavor supports this). If None, use
+        default schema.
+    if_exists : {'fail', 'replace', 'append'}, default 'fail'
+        How to behave if the table already exists.
+
+        * fail: Raise a ValueError.
+        * replace: Drop the table before inserting new values.
+        * append: Insert new values to the existing table.
+
+    index : bool, default True
+        Write DataFrame index as a column. Uses `index_label` as the column
+        name in the table.
+    index_label : str or sequence, default None
+        Column label for index column(s). If None is given (default) and
+        `index` is True, then the index names are used.
+        A sequence should be given if the DataFrame uses MultiIndex.
+    chunksize : int, optional
+        Specify the number of rows in each batch to be written at a time.
+        By default, all rows will be written at once.
+    dtype : dict or scalar, optional
+        Specifying the datatype for columns. If a dictionary is used, the
+        keys should be the column names and the values should be the
+        SQLAlchemy types or strings for the sqlite3 legacy mode. If a
+        scalar is provided, it will be applied to all columns.
+    compute : bool, default True
+        When true, call dask.compute and perform the load into SQL; otherwise, return a Dask object (or array of
+        per-block objects when parallel=True)
+    parallel : bool, default False
+        When true, have each block append itself to the DB table concurrently. This can result in DB rows being in a
+        different order than the source DataFrame's corresponding rows. When false, load each block into the SQL DB in
+        sequence.
+    kwargs : dict
+        If Pandas version is >= 0.24.0, a "method" value may be passed here; no other kwargs are allowed.
+
+        .. versionadded:: 0.24.0
```

Comment:
Just accept a `method` keyword here regardless of the version. Then if pandas <= 0.24.0 you can raise an exception if the default is not provided.

Then we don't have to validate that we aren't silently discarding arguments.